warehouse/accounts/services.py

```diff
@@ -255,8 +224,18 @@ def database_login_factory(context, request):
     )
 
 
-def user_token_factory(context, request):
-    return UserTokenService(
-        request.find_service(IUserService),
-        request.registry.settings
-    )
+class TokenServiceFactory:
+
+    def __init__(self, name, service_class=TokenService):
+        self.name = name
+        self.service_class = service_class
+
+    def __call__(self, context, request):
+        secret = request.registry.settings[f"token.{self.name}.secret"]
+        salt = self.name
```

Comment:
A little implicit for my tastes but not a problem. The first thing I looked for was appropriate differentiation of salts for each service, and it was kinda tough to find where that was happening.

Reply:
Yeah, I don't love this either. Do you think it makes sense to make the salt configurable and come from settings instead? Felt like that would be overkill. Open to other ideas here.

Reply:
Yeah, [itsdangerous recognizes the confusion this parameter name invokes](http://pythonhosted.org/itsdangerous/#the-salt) so it's probably best for us to have _some_ better name for it.

I'm by no means opposed to leaving this as is.